### PR TITLE
[tabulator-tables] add missing event callbacks + fix some function return types

### PR DESCRIPTION
**Pull request by lukecotter**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

**Review by Jojoshua — CHANGES_REQUESTED** [review]
Please see comments

File: types/tabulator-tables/index.d.ts [hunk 1]
```diff
@@ -2243,14 +2253,15 @@ export interface CellComponent {
 
 export interface EventCallBackMethods {
     validationFailed: (cell: CellComponent, value: any, validators: Validator[]) => void;
-    scrollHorizontal: (left: number) => void;
-    scrollVertical: (top: number) => void;
+    scrollHorizontal: (left: number, leftDir: boolean) => void;
```
[the remark this answers is not in the document]
Comment (lukecotter) [reply types/tabulator-tables/index.d.ts:2256]: Hi @Jojoshua 
Thank you for reviewing

I actually checked through all the `dispatchExternal` calls in the source for these.
This one is in the row [manager](https://github.com/olifolkerd/tabulator/blob/40f0a7fd649f7dd68a337ab2882f9fe142424fc8/src/js/core/RowManager.js#L130)

I am happy to remove, I was just trying to make things more accurate 😄

File: types/tabulator-tables/index.d.ts [hunk 2]
```diff
@@ -2341,13 +2356,12 @@ export interface EventCallBackMethods {
...
     cellMouseOut: (event: UIEvent, cell: CellComponent) => void;
     cellMouseMove: (event: UIEvent, cell: CellComponent) => void;
-    dataLoadError: (error: Error) => void;
-    dataProcessing: () => void;
-    dataProcessed: () => void;
     popupOpen: (cell: CellComponent) => void;
     popupClosed: (cell: CellComponent) => void;
     menuClosed: (cell: CellComponent) => void;
     menuOpened: (cell: CellComponent) => void;
+    TooltipClosed: (cell: CellComponent) => void;
```
[the remark this answers is not in the document]
Comment (lukecotter) [reply types/tabulator-tables/index.d.ts:2363]: I actually checked through all the `dispatchExternal` calls in the source for these.
Here it is in [Tooltip class](https://github.com/olifolkerd/tabulator/blob/f9a65455c9bb34ebe96f0136a201b5c446fc1d4e/src/js/modules/Tooltip/Tooltip.js#L119)

File: types/tabulator-tables/index.d.ts [hunk 3]
```diff
@@ -507,6 +516,7 @@ export interface OptionsData {
     dataLoader?: boolean;
     dataLoaderLoading?: string | HTMLElement;
     dataLoaderError?: string;
+    dataLoaderErrorTimeout?: number;
```
[the remark this answers is not in the document]
Comment (Jojoshua) [reply types/tabulator-tables/index.d.ts:519]: Ok, I have typically wanted to see public documentation but I am seeing where a lot of documentation is missing so thank you for the references

File: types/tabulator-tables/index.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jojoshua) [reply types/tabulator-tables/index.d.ts:2256]: Thank you for being thoughough!

**Review by Jojoshua — APPROVED** [review]

File: types/tabulator-tables/index.d.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (lukecotter) [reply types/tabulator-tables/index.d.ts:519]: no problem :)

File: types/tabulator-tables/index.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lukecotter) [reply types/tabulator-tables/index.d.ts:2256]: happy to contribute

**Comment by lukecotter** [comment]
Ready to merge

**Pull request merged**